crates/engine/src/prepare/trusted_documents.rs

```diff
@@ -278,22 +281,22 @@ async fn handle_trusted_document_query_and_check_that_query_matches<'ctx, 'r, R:
     document_id: Cow<'r, str>,
     inline_document: Option<&'r str>,
 ) -> Result<Cow<'r, str>, GraphqlError> {
-    let fetch_trusted_doc_result = handle_trusted_document_query(engine, client_name, document_id.clone()).await;
+    let document_id: String = document_id.into_owned();
+
+    let trusted_doc = match engine.runtime.operation_cache().get(&document_id).await {
```

Comment:
It's not the right cache key, we should have better types, this is the right one:
```
let cache_key = CacheKey::document(self.schema(), &DocumentKey::TrustedDocumentId {
    operation_name,
    client_name,
    doc_id,
});
```

Reply:
Good catch, just pushed that.